Gems/Atom/Tools/AtomToolsFramework/Code/Source/AssetSelection/AssetSelectionComboBox.cpp

```diff
@@ -41,29 +41,41 @@ namespace AtomToolsFramework
         AzFramework::AssetCatalogEventBus::Handler::BusDisconnect();
     }
 
-    void AssetSelectionComboBox::Reset()
+    void AssetSelectionComboBox::Clear()
     {
         clear();
         m_thumbnailKeys.clear();
+    }
+
+    void AssetSelectionComboBox::Populate()
+    {
+        Clear();
 
-        if (m_filterFn)
+        for (const auto& path : GetPathsInSourceFoldersMatchingFilter(m_filterFn))
         {
-            for (const auto& path : GetPathsInSourceFoldersMatchingFilter(m_filterFn))
-            {
-                AddPath(path);
-            }
-            setCurrentIndex(0);
+            AddPath(path);
```

Comment:
is it necessary to apply the filter first if AddPath also applies the filter?

Reply:
yes because the first function is building a container of files that match the filter, which would be every file in every gem/project if unfiltered, and AddPath can be run independently